src/plugins/graph/utilities/graph-utils.ts

```diff
@@ -140,37 +141,203 @@ export interface IMatchCirclesProps {
+  // dotsElement && select(dotsElement).on('click',
+  //   (event: MouseEvent) => {
+  //     const target = select(event.target as SVGSVGElement);
+  //     if (target.node()?.nodeName === 'circle') {
+  //       handleClickOnDot(event, (target.datum() as CaseData).caseID, dataConfiguration.dataset);
+  //     }
+  //   });
+  // dataConfiguration.setPointsNeedUpdating(false);
+}
+
+
+interface IInitializePoints {
+  selection: DotSelection;
+  data: any;
+  pointRadius: number;
+  pointColor: string;
+  pointStrokeColor: string;
+  getPointColorAtIndex?: (index: number) => string;
+  instanceId?: string;
+}
+
+const initializePoints = ({ selection, data, pointRadius,  pointColor,
+                            pointStrokeColor, getPointColorAtIndex, instanceId }: IInitializePoints) => {
+
+  //initialize outer highlight dots (this must be before the inner dots so that the inner dots are on top)
+  selection
+    .data(data)
+    .join(
+      (enter)=>
+        enter.append('circle')
+        .attr('class', 'graph-dot-highlighted')
+    );
+  //initialize inner dots
+  selection
+    .data(data)
     .join(
       (enter) =>
         enter.append('circle')
           .attr('class', 'graph-dot')
           .property('id', (d: CaseData) => `${instanceId}_${d.caseID}`),
-      (update) =>
-        update.attr('r', pointRadius)
-          .style('fill', pointColor)
-          .style('stroke', pointStrokeColor)
-          .style('stroke-width', defaultStrokeWidth)
     );
-  dotsElement && select(dotsElement).on('click',
-    (event: MouseEvent) => {
-      const target = select(event.target as SVGSVGElement);
-      if (target.node()?.nodeName === 'circle') {
-        handleClickOnDot(event, (target.datum() as CaseData).caseID, dataConfiguration.dataset);
+};
+
+export interface ISetPointSelection {
+  dotsRef: IDotsRef
+  dataConfiguration: IDataConfigurationModel
+  pointRadius: number,
+  selectedPointRadius: number,
+  pointColor: string,
+  pointStrokeColor: string,
+  getPointColorAtIndex?: (index: number) => string
+}
+
+
+export function setPointSelection(props: ISetPointSelection) {
+  const { dotsRef, dataConfiguration } = props;
+  const dataset = dataConfiguration.dataset;
+  const allCircles = selectAllCircles(dotsRef.current); //includes both inner and outer circles
+  const outerCircles = selectDots(dotsRef.current, true);
+  if (allCircles){
+    applySelectedClassToCircles(allCircles, dataset);
+  }
+  if (!(dotsRef.current && outerCircles)) return;
+  styleOuterCircles(outerCircles, dataset);
+}
+
+function applySelectedClassToCircles(selection: DotSelection, dataset?: IDataSet){
+  selection
+    .classed('selected', (aCaseData: CaseData) => !!dataset?.isCaseSelected(aCaseData.caseID));
+}
+
+function styleOuterCircles(outerCircles: any, dataset?: IDataSet){
+  if (dataset?.selection.size === 0) {
+    outerCircles
+      .attr('r', (aCaseData: CaseData) => {
+        return 0;
+      });
+  }
```

Comment:
This is case needed? Shouldn't `dataset?.isCaseSelected(aCaseData.caseID)` be `false` for all `caseID` if `dataset?.selection.size === 0`?

Reply:
The 
```
outerCircles
      .attr('r', (aCaseData: CaseData) => {
        return 0;
      });
```

Is called when we de-select (click outside the points) and `dataset.selection` has size 0, therefore we need to hide the outer circles.

Reply:
Try commenting out lines 233-239 and 251. Does it work any differently?

Reply:
I see what you mean - yeah it was unnecessary to perform that logic.

Reply:
Alright, best to remove it then. Including it is bad for two reasons:
- It duplicates code (the code that sets `r` to a hard coded `0`). It means that if we ever want to change that, it has to be changed in two places, which probably won't be obvious and could easily lead to bugs.
- It's confusing for engineers. Imagine someone unfamiliar with this code trying to understand it in a year from now. They will assume that the extra code is there for some important reason. Even if they're able to figure out that it's not necessary, it's going to take valuable (expensive) engineer time to solve that mystery.